src/crypto/tls/common.go

```diff
@@ -457,6 +477,11 @@ type ClientHelloInfo struct {
 	// Algorithms Extension is being used (see RFC 5246, Section 7.4.1.4.1).
 	SignatureSchemes []SignatureScheme
 
+	// SignatureSchemesDC lists the signature and hash schemes that the client
+	// is willing to verify when using delegated credentials.
+	// If Delegated Credentials are supported, this list should not be nil.
```

Comment:
Sorry, which spec?

Reply:
I mean the DC spec. From Sec. 4.1.1:

>   A client which supports this specification SHALL send a
>   "delegated_credential" extension in its ClientHello.  The body of the
>   extension consists of a SignatureSchemeList (defined in [RFC8446]):
>
>     struct {
>        SignatureScheme supported_signature_algorithm<2..2^16-2>;
>      } SignatureSchemeList;

Hence, the signature algorithm used for the delegated credential is negotiated separately from the signature algorithm used for the leaf certificate. I would just note this in the comment so that this is clear. For example:
>
Note that the signature algorithm used for delegated credentials is negotiated by the "delegated_credential" extension itself and not by the "signature_algorithms" extension.

Reply:
Ah yes ;)

Reply:
Actually, this made me think that it will be good to have the two lists for the supportedSignatureAlgorithms after all ;)

Reply:
Fixed now.